crates/fuel-core/src/schema/tx/types.rs

```diff
@@ -272,6 +311,47 @@ impl FailureStatus {
     }
 }
 
+#[derive(Debug)]
+pub struct FailureDuringBlockProductionStatus {
+    pub tx_pointer: TxPointer,
+    pub tx_id: Option<TxId>,
+    pub receipts: Option<Vec<fuel_tx::Receipt>>,
+    pub reason: String,
+}
+
+#[Object]
+impl FailureDuringBlockProductionStatus {
+    async fn reason(&self) -> String {
+        self.reason.clone()
+    }
+
+    async fn tx_pointer(&self) -> TxPointer {
+        self.tx_pointer
+    }
+
+    async fn transaction_id(&self) -> Option<TransactionId> {
+        self.tx_id.map(Into::into)
+    }
+
+    #[graphql(complexity = "query_costs().storage_read + child_complexity")]
+    async fn transaction(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Option<async_graphql::Result<Transaction>> {
+        self.tx_id.map(|tx_id| {
+            let query = ctx.read_view()?;
+            let transaction = query.transaction(&tx_id)?;
+            Ok(Transaction::from_tx(tx_id, transaction))
+        })
+    }
+
+    async fn receipts(&self) -> Option<async_graphql::Result<Vec<Receipt>>> {
```

Comment:
Why do we need to return an error? Looks like we can't have error here

Reply:
I did it like that for consistency with already existing types, like `SuccessStatus` [here](https://github.com/FuelLabs/fuel-core/blob/master/crates/fuel-core/src/schema/tx/types.rs#L202-L204).